core/blockchain.go

```diff
@@ -963,12 +964,21 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			events = append(events, ChainSplitEvent{block, logs})
 		}
 		stats.processed++
+
+		// Update the current gas limit (mining only) based on import times
```

Comment:
I think it would make sense to divide by gas used, so it's measuring how quickly/slowly gas is consumed, rather than how long the block takes.